### ISSUE TITLE
AWS::CodePipeline::Pipeline strict parameters

### ISSUE THREAD

**Issue by deleugpn** [aws-cloudformation/cfn-python-lint#1516]
*cfn-lint version: 0.30.1*

*Description of issue.*

The `Version` attribute of `Properties.Stages.Actions.ActionTypeId` is clearly a number, but is required to be specified as String. 

```
  ProxyPipeline:
    Type: AWS::CodePipeline::Pipeline
    Properties:
      ArtifactStore:
        Location: !ImportValue SourceCodeStorage
        Type: S3
      Name: !Sub "${Environment}-proxy"
      RestartExecutionOnUpdate: false
      RoleArn: !ImportValue CodePipelineRoleArn
      Stages:
        - Name: Source
          Actions:
            - Name: Source
              ActionTypeId:
                Category: Source
                Owner: ThirdParty
                Provider: GitHub
                Version: 1
              Configuration:
                Owner: MyOrganization
                Repo: MyRepository
                Branch: Mybranch
                OAuthToken: !ImportValue GitHubOAuthToken
                PollForSourceChanges: false
              OutputArtifacts:
                - Name: source-code
```

**Comment by deleugpn** [comment]
Coming back to this one to leave more details. Apparently the issue only happens for `Source` and `Build` stage, but not for Deployment Stage. In the following template, the first 2 `Version` attributes are reported and the last 2 are **not** reported

```
  ProxyPipeline:
    Type: AWS::CodePipeline::Pipeline
    Properties:
      ArtifactStore:
        Location: !ImportValue SourceCodeStorage
        Type: S3
      Name: !Sub "${Environment}-proxy"
      RestartExecutionOnUpdate: false
      RoleArn: !ImportValue CodePipelineRoleArn
      Stages:
        - Name: Source
          Actions:
            - Name: Source
              ActionTypeId:
                Category: Source
                Owner: ThirdParty
                Provider: GitHub
                Version: 1          # REPORTED
              Configuration:
                Owner: MyOrg
                Repo: MyRepo
                Branch: MyBranch
                OAuthToken: !ImportValue GitHubOAuthToken
                PollForSourceChanges: false
              OutputArtifacts:
                - Name: source-code

        - Name: Build
          Actions:
            - Name: CodeBuild
              ActionTypeId:
                Category: Build
                Owner: AWS
                Provider: CodeBuild
                Version: 1        # REPORTED
              Configuration:
                ProjectName: !Ref CodeBuild
                PrimarySource: source_code
              InputArtifacts:
                - Name: source_code
              OutputArtifacts:
                - Name: build

        - Name: Deploy
          Actions:
            - Name: Infrastructure
              InputArtifacts:
                - Name: source-code
              ActionTypeId:
                Category: Deploy
                Owner: AWS
                Version: 1       # NOT REPORTED!
                Provider: CloudFormation
              OutputArtifacts: []
              Configuration:
                StackName: !Sub ${Environment}-proxy-infrastructure
                ActionMode: CREATE_UPDATE
                RoleArn: !ImportValue CloudFormationRoleArn
                TemplatePath: source-code::aws/infrastructure/template.yaml
                ParameterOverrides: !Sub |
                  {
                    "Environment": "${Environment}"
                  }
              RunOrder: 1

            - Name: Proxy
              InputArtifacts:
                - Name: source-code
              ActionTypeId:
                Category: Deploy
                Owner: AWS
                Version: 1       # NOT REPORTED!
                Provider: CloudFormation
              OutputArtifacts: []
              Configuration:
                StackName: !Sub ${Environment}-proxy-service
                ActionMode: CREATE_UPDATE
                RoleArn: !ImportValue CloudFormationRoleArn
                TemplatePath: source-code::aws/deploy/template.yaml
                TemplateConfiguration: build::image.json
                ParameterOverrides: !Sub |
                  {
                    "Environment": "${Environment}"
                  }
              RunOrder: 2
```

**Comment by PatMyron** [comment]
duplicate of https://github.com/aws-cloudformation/cfn-python-lint/issues/547
